test/gtest/common/test.cc

```diff
@@ -123,23 +123,26 @@ void test_base::pop_config()
     m_config_stack.pop_back();
 }
 
-void test_base::hide_errors()
-{
-    ucs_log_push_handler(hide_errors_logger);
-}
-
-void test_base::hide_warnings()
-{
-    ucs_log_push_handler(hide_warns_logger);
-}
-
-void test_base::wrap_errors()
-{
-    ucs_log_push_handler(wrap_errors_logger);
+test_base::scoped_log_handler::scoped_log_handler(scoped_log_handler_type_t type) {
```

Comment:
maybe pass handler func ptr as a pointer? would not need switch/case, and avoid defining specific function like sockaddr_errs_detector in base class

Reply:
maybe also add new C++ and use lambda? For example smth like 

```
    ucs_status_t status;

    {
        scoped_log_handler hide_err(hide_errors_logger);
        status = ucp_atomic_add64(e->ep(), 0, (uintptr_t)memheap_addr + 1, rkey);
    }

    EXPECT_EQ(UCS_ERR_INVALID_PARAM, status);
```
may be replaced with smth like 
```
    ucs_status_t status = [] (ucs_log_func_t log_func) {
        scoped_log_handler hide_err(log_func);
        return ucp_atomic_add64(e->ep(), 0, (uintptr_t)memheap_addr + 1, rkey);
    } (hide_errors_logger);

    EXPECT_EQ(UCS_ERR_INVALID_PARAM, status);
```

where `hide_errors_logger` also may be a local lambda when it's needed

Reply:
i would like that, but today we don't require c11